# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: 339795d0a51e
# Full Hash: 339795d0a51eb84d7d792e1b610c068f52f99b7a
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-04-21 09:42:20
# Regressor Bug: 1574746
# File Overlap Count: 1
# Description:
#   Bug 1574746 - Remove AllowWebRenderForThisWindow() r=nical
#   
#   nsBaseWidget::AllowWebRenderForThisWindow() limits WebRender usage to some  type of windows. It was added by Bug 1377321, because initialization of WebRender was heavy weight in the past. It is not necessary anymore.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71440
# ==============================================================================

diff -r 0e0b2528287a -r 339795d0a51e widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Mon Apr 20 20:32:27 2020 +0000
+++ b/widget/gtk/nsWindow.cpp	Mon Apr 20 14:15:08 2020 +0000
@@ -3963,10 +3963,8 @@
       // gfxVars, used below.
       Unused << gfxPlatform::GetPlatform();
 
-      bool useWebRender =
-          gfx::gfxVars::UseWebRender() && AllowWebRenderForThisWindow();
-
       mIsAccelerated = ComputeShouldAccelerate();
+      bool useWebRender = gfx::gfxVars::UseWebRender() && mIsAccelerated;
       MOZ_ASSERT(mIsAccelerated | !useWebRender);
 
       if (mWindowType == eWindowType_toplevel) {